src/temporal/common.ts

```diff
@@ -21,3 +24,35 @@ export const generateWorkflowId = (
 export interface InjectedProps {
   con: DataSource;
 }
+
+export enum TemporalError {
+  NotFound = 'WorkflowNotFoundError',
+}
+
+export const getWorkflowHandle = async (
+  workflowId: string,
+): Promise<WorkflowHandle> => {
+  const client = await getTemporalClient();
+
+  return client.workflow.getHandle(workflowId);
+};
+
+export const getDescribeOrError = (handle: WorkflowHandle) => {
+  try {
+    return handle.describe();
+  } catch (error) {
+    if (error.name === TemporalError.NotFound) {
+      return;
+    }
+
+    throw error;
+  }
+};
+
+export const getWorkflowDescription = async (
+  workflowId: string,
+): Promise<WorkflowExecutionDescription> => {
```

Comment:
This then returns workflow description or undefined (in case of NotFound) right?

Reply:
Yes. Should we explicitly say `| undefined` as part of the TS strict mode?

Reply:
I think so yes